cirq/google/engine/engine_program.py

```diff
@@ -112,7 +131,188 @@ def run(
+            context.Pack(
+                v1.program_pb2.RunContext(parameter_sweeps=[
+                    v1.sweep_to_proto(sweep, repetitions) for sweep in sweeps
+                ]))
+        elif proto_version == engine_base.ProtoVersion.V2:
+            run_context = v2.run_context_pb2.RunContext()
+            for sweep in sweeps:
+                sweep_proto = run_context.parameter_sweeps.add()
+                sweep_proto.repetitions = repetitions
+                v2.sweep_to_proto(sweep, out=sweep_proto.sweep)
+
+            context.Pack(run_context)
+        else:
+            raise ValueError(
+                'invalid run context proto version: {}'.format(proto_version))
+        return context
+
+    def get_circuit(self) -> 'Circuit':
+        """Returns the cirq Circuit for the Quantum Engine program. This is only
+        supported if the program was created with the V2 protos.
+
+        Returns:
+            The program's cirq Circuit.
+        """
+        if not self._program.HasField('code'):
+            self._program = self._engine.client.get_program(
+                self.project_id, self.program_id, True)
+        return self._deserialize_program(self._program.code)
+
+    def _deserialize_program(self, code: qtypes.any_pb2.Any) -> 'Circuit':
+        code_type = code.type_url[len(engine_base.TYPE_PREFIX):]
+        if (code_type == 'cirq.google.api.v1.Program' or
+                code_type == 'cirq.api.google.v1.Program'):
+            raise ValueError('deserializing a v1 Program is not supported')
+        if (code_type == 'cirq.google.api.v2.Program' or
+                code_type == 'cirq.api.google.v2.Program'):
+            program = v2.program_pb2.Program()
+            program.ParseFromString(code.value)
+            gate_set_map = {
+                g.gate_set_name: g for g in gate_sets.GOOGLE_GATESETS
+            }
+            try:
+                return gate_set_map[program.language.gate_set].deserialize(
+                    program)
+            except KeyError:
+                raise ValueError('unsupported gateset: {}'.format(
+                    program.language.gate_set))
+        raise ValueError('unsupported program type: {}'.format(code_type))
+
+    def description(self) -> str:
+        """Returns the description of the program.
+
+        Returns:
+             The current description of the program.
+        """
+        return self._program.description
+
+    def set_description(self, description: str) -> 'EngineProgram':
+        """Sets the description of the program.
+
+        Params:
+            description: The new description for the program.
+
+        Returns:
+             This EngineProgram.
+        """
+        self._program = self._engine.client.set_program_description(
+            self.project_id, self.program_id, description)
+        return self
+
+    def labels(self) -> Dict[str, str]:
+        """Returns the labels of the program.
+
+        Returns:
```

Comment:
This is redundant.  I'd have one or the other.

Reply:
Removed.